addons/mail/models/mail_canned_response.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class MailCannedResponse(models.Model):
+    """
+    Canned Response: content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.
+    """
+
+    _name = "mail.canned_response"
+    _description = "Canned Response"
+    source = fields.Char(
+        "Shortcut",
+        required=True,
+        index="trigram",
+        help="Canned response that will automatically be substituted with longer content in your messages."
+        " Type ':' followed by the name of your shortcut (e.g. :hello) to use in your messages.",
+    )
+    substitution = fields.Text(
+        "Substitution",
+        required=True,
+        help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.",
+    )
+    description = fields.Char("Description")
+    last_used = fields.Datetime("Last Used", help="Last time this canned_response was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="mail.canned_response_administrator")
+    is_shared = fields.Boolean(
+        string="Determines if the canned_response is currently shared with other users",
+        compute="_compute_is_shared",
+        store=True,
+    )
+    is_editable = fields.Boolean(
+        string="Determines if the canned response can be edited by the current user", compute="_compute_is_editable"
+    )
+
+    @api.depends("create_uid")
+    def _compute_is_editable(self):
+        for canned_response in self:
+            canned_response.is_editable = bool(
+                self.env.is_admin() or canned_response.create_uid.id == self.env.uid or not canned_response.id
```

Comment:
Shouldn't it check for the admin group rather than `is_admin`?

Reply:
It look like is_admin is doing that https://github.com/odoo-dev/odoo/blob/master-mail-rework-canned-response-did/odoo/addons/base/models/res_users.py#L1124
But maybe you mean an other group?